src/robot/running/context.py

```diff
@@ -13,15 +13,36 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+import asyncio
 from contextlib import contextmanager
 
 from robot.errors import DataError
 
 
+class _EventLoop:
+
+    def __init__(self) -> None:
+        self._loop_ref = None
+    
+    @property
+    def running_event_loop(self):
+        if self._loop_ref is None:
+            self._loop_ref = asyncio.new_event_loop()
+        return self._loop_ref
+
+    def run_until_complete(self, coro):
+        return self.running_event_loop.run_until_complete(coro)
+
+    def close_loop(self):
+        if self._loop_ref:
+            self._loop_ref.close()
```

Comment:
An explicit class for handling the event loop is a good idea. No need for it to start with `_`, though. We used that style earlier to indicate a class was private not not anymore.